File: PROGRAMMERS/lv0/javascript/구슬을 나누는 경우의 수.js
let fac = (n) => (n == 0 ? 1 : n * fac(n - 1));

function solution(balls, share) {
    return Math.round(fac(balls) / fac(balls - share) / fac(share));
}

let inps = [
    [3, 2],
    [5, 3],
];

for (let i of inps) {
    console.log(solution(i[0], i[1]));
}

File: PROGRAMMERS/lv0/python/구슬을 나누는 경우의 수.py
def solution(balls, share):
    ans = 1

    for v in range(1, balls+1):
        ans *= v
        if(v <= balls-share): ans /= v
        if(v <= share): ans /= v

    return int(ans)

inps = [
    [3, 2],
    [5, 3],
    [1, 1]
]

for (b, s) in inps:
    print(solution(b, s))